Tuesdays/Week-Class/Inheratance/Assets/Scripts/PlayableCharacter.cs

```diff
@@ -6,21 +6,24 @@ public abstract class PlayableCharacter : MonoBehaviour, IDamagable{
 	[SerializeField] int speed;
 	[SerializeField] int maxHP;
 	int currHP;
+	public delegate void HPDelegate(int hp);
+	public event HPDelegate HPChangeHandler; 
 
+	public int Speed { get { return speed; } }
+	public int CurrHP { get { return currHP; } }
+	
 	private void Awake() {
 		currHP = maxHP;
 	}
 
-
-	public int Speed { get { return speed; } }
-	public int CurrHP { get { return currHP; } }
 	public void Die() {
 		this.gameObject.SetActive(false);
 		this.gameObject.GetComponentInParent<PlayerController>().SwitchCharacter();
 	}
 
 	public void TakeDamage(int howMuch) {
 		currHP -= howMuch;
+		HPChangeHandler(currHP); //Raising the event
```

Comment:
you wanted to call `.Invoke()` with the right parameters
not sure if it works that way